### PR TITLE
WIP: Rhoop/jenkins fixes

### PR DESCRIPTION
**Pull request by rhoop**
# DO NOT MERGE YET

# Goals of PR

### Core changes: 
Getting docker compose to work for devnet

### Notes: 
Lots of work on scripts and build env

### CODE REVIEW

File: docker/sp-geth.Dockerfile [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    mv cast /usr/local/bin
+
+# copy everything we need
+WORKDIR /specular
+
+RUN mkdir -p /specular/workspace
+RUN mkdir -p /specular/sbin
+RUN mkdir -p /specular/contracts
+
+# install hardhar
```
Comment (chalkandpaste) [inline docker/sp-geth.Dockerfile:29]: hardy-har... =P

File: sbin/create_genesis.sh [hunk 2]
```diff
@@ -5,11 +5,12 @@ SBIN="$(
   cd "$SBIN"
   pwd
 )"
-. $SBIN/utils/utils.sh
+. "$SBIN/utils/utils.sh"
 ROOT_DIR=$SBIN/..
 
 # Check that the all required dotenv files exists.
 reqdotenv "paths" ".paths.env"
+source .paths.env
```
Comment (chalkandpaste) [inline sbin/create_genesis.sh:13]: What's going on here? Doesn't `reqdotenv` already source?

**Review by chalkandpaste — APPROVED** [review]
Aside from the one-off comments, LGTM

File: sbin/create_genesis.sh (hunk 2 above)
Comment (rhoop) [reply sbin/create_genesis.sh:13]: This is a debug line I am testing something out. It will be removed

File: docker/sp-geth.Dockerfile (hunk 1 above)
Comment (rhoop) [reply docker/sp-geth.Dockerfile:29]: 🤦

File: sbin/utils/utils.sh [hunk 3]
```diff
@@ -15,13 +15,26 @@ guard() {
 
 # Requests a user to confirm that overwriting
 # file ($1) is okay, if it exists.
+# doesn't ask if -y is set
 guard_overwrite() {
+
   if test -f $1; then
+<<<<<<< HEAD
```
Comment (macwis) [inline sbin/utils/utils.sh:22]: merge artifact

**Pull request closed without merging**